### PR TITLE
improve perf for markdown.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs [hunk 1]
```diff
@@ -49,5 +73,19 @@ private static int GetValidLineCount(string markdown)
...
         }
+
+        private static int GetValidLineCount(string markdown, int lineCount)
+        {
+            if (markdown == "")
+                return 0;
+            var indexOfLastChar = markdown.Length - 1;
+            var validLineCount = lineCount;
+
+            while (indexOfLastChar >= 0 && markdown[indexOfLastChar--] == '\n')
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs]: >'\n' [](start = 74, length = 4)

all "\r\n" are normalized to "\n" already?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs]: yes.

File: src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs [hunk 2]
```diff
@@ -49,5 +73,19 @@ private static int GetValidLineCount(string markdown)
 
             return validLineCount;
         }
+
+        private static int GetValidLineCount(string markdown, int lineCount)
+        {
+            if (markdown == "")
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs]: >"" [](start = 28, length = 2)

minor..string.empty?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/SourceInfo.cs]: fixed.

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**